util/webhook/webhook.go

```diff
@@ -143,20 +188,36 @@ func (a *ArgoCDWebhookHandler) HandleEvent(payload interface{}, header webhooks.
 }
 
 func (a *ArgoCDWebhookHandler) Handler(w http.ResponseWriter, r *http.Request) {
-	event := r.Header.Get("X-GitHub-Event")
-	if len(event) > 0 {
-		a.githubHandler.ServeHTTP(w, r)
-		return
-	}
-	event = r.Header.Get("X-Gitlab-Event")
-	if len(event) > 0 {
-		a.gitlabHandler.ServeHTTP(w, r)
+
+	var payload interface{}
+	var err error
+
+	switch {
+	//Gogs needs to be checked before Github since it carries both Gogs and (incompatible) Github headers
```

Comment:
good comment

Reply:
What a shame Gogs injects incompatible Github header (missing some fields, so the Github module crashed to retrieve the "createdAt"). Otherwise it would have been supported directly without Gogs implementation.